tests/integration/testlib/statelib.py

```diff
@@ -117,13 +118,30 @@ def normalize(self):
         self._ignore_ipv6_link_local()
         self._sort_ip_addresses()
         self._sort_interfaces_by_name()
+        self.sanitize_ipv6()
 
     def match(self, other):
         return _state_match(self.state, other.state)
 
     def _sort_interfaces_by_name(self):
         self._state[INTERFACES].sort(key=lambda d: d['name'])
 
+    def sanitize_ipv6(self):
+        for iface_state in self.state[Interface.KEY]:
+            if Interface.IPV6 in iface_state:
+                iface_ipv6_state = iface_state[Interface.IPV6]
+                if InterfaceIP.ADDRESS in iface_ipv6_state:
+                    iface_ipv6_addresses_state = iface_ipv6_state[
+                        InterfaceIP.ADDRESS
+                    ]
+                    compressed_ipv6_addrs = [
+                        compress_ipv6_addr(iface_ipv6_addr)
+                        for iface_ipv6_addr in iface_ipv6_addresses_state
+                    ]
+                    iface_ipv6_state[
+                        InterfaceIP.ADDRESS
+                    ] = compressed_ipv6_addrs
```

Comment:
Can you please check if this will read better?
```
iface_ipv6_state[InterfaceIP.ADDRESS] = [
    compress_ipv6_addr(iface_ipv6_addr)
    for iface_ipv6_addr in iface_ipv6_addresses_state
]
```

Reply:
could be, at least it makes better use of the 80 char line limit.

I'll update - removing the variable, since we never re-use it.